src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -0,0 +1,386 @@
+      - "trapezoid" for the trapezoidal rule
+      - "simpson" for the Simpson's rule (for unequally spaced points, Shklov, 1960)
+      - "intensity_sum" for the simple sum of the intensities
+
+      @note Make sure the spectrum is sorted with respect to mass-to-charge ratio.
+
+      @param[in] spectrum The spectrum which contains the peak
+      @param[in] left The left mass-to-charge ratio boundary
+      @param[in] right The right mass-to-charge ratio boundary
+    */
+    void integratePeak(const MSSpectrum& spectrum, const double& left, const double& right);
+
+    /**
+      @brief Calculate peak's shape metrics.
+
+      The calculated characteristics are the start and end times at 0.05, 0.10 and
+      0.5 the peak's height. Also the widths at those positions are calculated.
+      Other values: the peak's total width, its tailing factor, asymmetry factor,
+      baseline delta to height and the slope of the baseline.
+      The number of points across the baseline and also at half height are saved.
+
+      @note Make sure the chromatogram is sorted with respect to retention time.
+
+      @param[in] chromatogram The chromatogram which contains the peak
+      @param[in] left The left retention time boundary
+      @param[in] right The right retention time boundary
+    */
+    void calculatePeakShapeMetrics(const MSChromatogram& chromatogram, const double& left, const double& right);
+    /**
+      @brief Calculate peak's shape metrics.
+
+      The calculated characteristics are the start and end times at 0.05, 0.10 and
+      0.5 the peak's height. Also the widths at those positions are calculated.
+      Other values: the peak's total width, its tailing factor, asymmetry factor,
+      baseline delta to height and the slope of the baseline.
+      The number of points across the baseline and also at half height are saved.
+
+      @note Make sure the spectrum is sorted with respect to mass-to-charge ratio.
+
+      @param[in] spectrum The spectrum which contains the peak
+      @param[in] left The left mass-to-charge ratio boundary
+      @param[in] right The right mass-to-charge ratio boundary
+    */
+    void calculatePeakShapeMetrics(const MSSpectrum& spectrum, const double& left, const double& right);
+
+    void getDefaultParameters(Param& params);
+
+    /** @name calculatePeakShapeMetrics() outputs
+      The calculatePeakShapeMetrics() method uses this struct to save its results.
+    */
+    ///@{
+    struct PeakShapeMetrics
+    {
+      /**
+        The width of the peak at 5% the peak's height.
+      */
+      double width_at_5 = 0.0;
+      /**
+        The width of the peak at 10% the peak's height.
+      */
+      double width_at_10 = 0.0;
+      /**
+        The width of the peak at 50% the peak's height.
+      */
+      double width_at_50 = 0.0;
+      /**
+        The start position at which the intensity is 5% the peak's height.
+      */
+      double start_time_at_5 = 0.0;
```

Comment:
time might be a misnomer if metrics are calculated for mass peaks (works only for chromatographic peaks)